core/trino-main/src/main/java/io/trino/execution/scheduler/EventDrivenFaultTolerantQueryScheduler.java

```diff
@@ -1525,6 +1526,12 @@ public void addPartition(int partitionId, NodeRequirements nodeRequirements)
             DataSize defaultTaskMemory = stage.getFragment().getPartitioning().equals(COORDINATOR_DISTRIBUTION) ?
                     getFaultTolerantExecutionDefaultCoordinatorTaskMemory(session) :
                     getFaultTolerantExecutionDefaultTaskMemory(session);
+            if (stage.getFragment().getRoot().getSources().stream()
+                    .anyMatch(planNode -> planNode instanceof RefreshMaterializedViewNode)) {
```

Comment:
Will this condition match for a REFRESH MV on iceberg MVs as well ? I think that can execute the query within the same sql command

Reply:
IIUC this plan node translates to `RefreshMaterializedViewOperator` which under transitively calls out to `ConnectorMetadata.refreshMaterializedView`. Is this the codepath you are referring to?

Reply:
> Will this condition match for a REFRESH MV on iceberg MVs as well ? I think that can execute the query within the same sql command

no. `RefreshMaterializedViewNode` and corresponding operator is used only when MV refresh is delegated to connector.

```
        if (delegatedRefreshMaterializedView.isPresent()) {
            return new RelationPlan(
                    new RefreshMaterializedViewNode(idAllocator.getNextId(), delegatedRefreshMaterializedView.get()),
                    analysis.getRootScope(),
                    ImmutableList.of(),
                    Optional.empty());
        }

```

Reply:
got it, this is correct then